var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -73,6 +73,14 @@ class Cuda(Package):
     # Mojave support -- only macOS High Sierra 10.13 is supported.
     conflicts('arch=darwin-mojave-x86_64')
 
+    depends_on('libxml2', type='build', when='@10.1.243:')
+
+    def setup_build_environment(self, env):
+        if self.spec.satisfies('@10.1.243:'):
+            libxml2_home  = self.spec['libxml2'].prefix
+            env.set('LIBXML2HOME', libxml2_home)
+            env.append_path('LD_LIBRARY_PATH', libxml2_home.lib)
```

Comment:
If this is needed for linking, you should change the deptype above. Then Spack will add the linker flags automatically, and this line may not even be necessary.

Reply:
I'm not personally familiar with this package, and to me It's a little opaque what `libxml2` is needed for.  This is a binary blob installer, where it just downloads a `.run` file and executes it.  You'd think the `libxml2` dep is needed for linking, based on the error message, but I tried with `type=('build', 'link', 'run')` (and not manually setting `LD_LIBRARY_PATH`) and it was still broken.

So to my understanding it still needs the `setup_build_environment()` method.  But I'm not sure what's necessary and sufficient as far as the dep type goes.  In that case, is it safest to make it `('build', 'link', 'run')`?

Thanks for the review.

Reply:
Hmm, I guess that means it isn't using Spack's compiler wrappers. I think if you leave it as the default (build/link) that'll be fine.

Reply:
Just my two cents: as @scottwittenburg pointed out, the `.run` installer is a static binary that decompress some binary blob that itself contains something that needs to dyload `libxml2`. I guess that `LD_LIBRARY_PATH` is the only way to make this absurd installer happy.